### PR TITLE
feat(staking,common,extension): add details drawer in staking

### PR DESCRIPTION
**Pull request by przemyslaw-wlodek**
# Checklist

- [x] JIRA - LW-6678
- [x] Screenshots added.

---

## Proposed solution

This PR reintroduces the pool details drawer in the staking package and necessary refactoring for common components e.g. Banner.

## Testing

1. Open Staking
2. Go to the **Browse Pools** tab
3. Click on a Pool
4. Observe the Stake Pool Details Drawer being displayed

## Screenshots
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/684/5409477544/index.html) for [8f0b9557](https://github.com/input-output-hk/lace/pull/193/commits/8f0b95579f4b9c9625224e726f898cf6ef2fcaa5)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 34     | 2      | 0       | 0     | 36    | ❌     |

### CODE REVIEW

**Review by mrcnk — APPROVED** [review]
Looks good 👍

File: packages/staking/src/features/outside-handles-provider/OutsideHandlesProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  password,
+  passwordRemovePassword,
+  submittingStateSetIsRestaking,
+  walletStoreGetKeyAgentType,
+  walletStoreInMemoryWallet,
+  walletStoreWalletUICardanoCoin,
+}: OutsideHandlesProviderProps) => {
+  const contextValue = useMemo<OutsideHandlesContextValue>(
+    () => ({
+      backgroundServiceAPIContextSetWalletPassword,
```
Comment (danielmain) [inline packages/staking/src/features/outside-handles-provider/OutsideHandlesProvider.tsx:24]: Create a variable with all this properties and pass it in the array in line 37 with `...object` with that you will save a ton of lines of code :)

File: packages/staking/src/features/overview/Overview.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ export const Overview = () => {
             {...item}
             markerColor={item.color}
             cardanoCoinSymbol={'tADA'}
-            onStakePoolSelect={() => console.log('onStakePoolSelect')}
+            onStakePoolSelect={() => console.info('onStakePoolSelect')}
```
Comment (danielmain) [inline packages/staking/src/features/overview/Overview.tsx:66]: I think console does not belong in production code

**Review by danielmain — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/features/delegation/components/DelegationContainer.tsx [hunk 3]
```diff
@@ -5,14 +5,7 @@ import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import isNumber from 'lodash/isNumber';
 import { Wallet } from '@lace/cardano';
 import { walletRoutePaths } from '@routes';
-import {
-  useRedirection,
```
[the remark this answers is not in the document]
Comment (przemyslaw-wlodek) [reply apps/browser-extension-wallet/src/features/delegation/components/DelegationContainer.tsx:9]: Hey :) Thank you for taking the time to review this PR.

This code is mostly like lift and shift. That was our squad agreement for now, as it requires a lot of work on our side to untangle the providers/hooks and everything that was used there in the browser extension. Most of the stuff will be addressed in future PRs (like console.logs and any other TODO/placeholders).

There are no new formatting or linting issues in this PR - we have a common configuration and it's checked in the pipeline.

I really hope that we can refactor this in the incoming sprint and PI. :)

**Review by danielmain — APPROVED** [review]

**Review by ManusMcCole — APPROVED** [review]
Looks good @mrcnk 👍

**Pull request merged**